#130341359 Redesign UI for gaming screen

demiladebamgbose opened the pull request:
#### What does this PR do?

Redesigns a new functional UI for the gaming screen.
#### Description of Task to be completed?
- Design UI
- Plugin functionalities to play a game
#### How should this be manually tested?

When Users click `PLAY` or `PLAY NOW` from the landing page, they are directed to the new gaming screen which is a redesigned interface to play the game.
#### What are the relevant pivotal tracker stories?
#130341359 Redesign UI for the actual gaming screen
#### Screenshots (if appropriate)

![screenshot from 2016-10-18 23-16-54](https://cloud.githubusercontent.com/assets/17247842/19498591/c40f18e4-9589-11e6-80bb-110907fe1b1c.png)

![screenshot from 2016-10-18 23-13-38](https://cloud.githubusercontent.com/assets/17247842/19498592/c5644354-9589-11e6-954c-f9936bbd509e.png)

![screenshot from 2016-10-18 23-13-55](https://cloud.githubusercontent.com/assets/17247842/19498605/d12d4d16-9589-11e6-9b0f-4e2b1fe5b6e8.png)

![screenshot from 2016-10-18 23-14-55](https://cloud.githubusercontent.com/assets/17247842/19498608/d85072f8-9589-11e6-93c5-8515743ea891.png)

demiladebamgbose committed:
Feature: Design static html

Design static gaming page with html and css

Start #130341359

demiladebamgbose committed:
Feature: Design static html

Design static gaming page with html and css

Start #130341359

demiladebamgbose committed:
Feature: Design static html

Design static gaming page with html and css

Start #130341359

demiladebamgbose committed:
Feature: Design static html

Design static gaming page with html and css

Start #130341359

demiladebamgbose committed:
Feature: Design static html

Design static gaming page with html and css

Start #130341359

demiladebamgbose committed:
Merge branch 'feature/#130341359/re-design-ui-for-gaming-screen' of https://github.com/andela/project-kinsan into feature/#130341359/re-design-ui-for-gaming-screen

demiladebamgbose committed:
bug: remove cunflicting css

remove conflicting css

start #130341359

demiladebamgbose committed:
bug: remove cunflicting css

remove conflicting css

start #130341359

demiladebamgbose committed:
bug: remove cunflicting css

remove conflicting css

start #130341359

demiladebamgbose committed:
bug: remove cunflicting css

remove conflicting css

start #130341359

Commit:
Merge branch 'feature/#130341359/re-design-ui-for-gaming-screen' of github.com:andela/project-kinsan into feature/#130341359/re-design-ui-for-gaming-screen

Commit:
Feature(update-angular): Update angular in foot.jade to 1.5.8

Commit:
Feature(update-angular): Update angular in foot.jade to 1.5.8

Commit:
Feature(add-angular-route): Add angular-route to the project

Due to the upgrade of angular from 1.1.5 to 1.5.8 it becomes
necessary to add angular-route package to the project because as
angular 1.2 angularroute had been moved to its own package.

Commit:
Feature(add-angular-route): Add angular-route to the project

Due to the upgrade of angular from 1.1.5 to 1.5.8 it becomes
necessary to add angular-route package to the project because as
angular 1.2 angularroute had been moved to its own package.

Commit:
Feature(add-ng-html-js): Add ng-html2js package to help in testing directives with template url

Commit:
Feature(add-ng-html-js): Add ng-html2js package to help in testing directives with template url

Commit:
Feature(add-ng-html2js-to-karma): Add ng-html2js preprocessor to karma

Commit:
Feature(add-ng-html2js-to-karma): Add ng-html2js preprocessor to karma

Commit:
Feature(player-test): Add test for player component

Commit:
Feature(player-test): Add test for player component

Commit:
Feature(test): Modify test files and karma to drop public in link to teplate html file

Commit:
Feature(test): Modify test files and karma to drop public in link to teplate html file

Commit:
Feature(add-controller): Add GameController to game.tpl.html template

Commit:
Feature(add-controller): Add GameController to game.tpl.html template

Commit:
Feature(Player-component): Implemet player component to display players on the game screen

Commit:
Feature(Player-component): Implemet player component to display players on the game screen

Commit:
Feature(add-is-czar): Add a label to show who is czar per round

Commit:
Feature(add-is-czar): Add a label to show who is czar per round

Commit:
Feature(fix-css): Fix name styling for displaying the name

Commit:
Feature(fix-css): Fix name styling for displaying the name

Commit:
Feature(is-czar): Add function to player controller to show czar

Commit:
Feature(is-czar): Add function to player controller to show czar

Commit:
Feature(modify-css): Modify size of cards and isCzar style

Make the size of cards in the game smaller to accommodate more elements.
Also modify the isCzar option for the players section.

Commit:
Feature(modify-css): Modify size of cards and isCzar style

Make the size of cards in the game smaller to accommodate more elements.
Also modify the isCzar option for the players section.

Commit:
Feature(add-timer-question-tag): Add the question and timer tag for their respective components to the game screen

Commit:
Feature(add-timer-question-tag): Add the question and timer tag for their respective components to the game screen

Commit:
Feature(question-template): Implement the question component template

Commit:
Feature(question-template): Implement the question component template

Commit:
Feature(timer-template): Implement the timer component template

Commit:
Feature(timer-template): Implement the timer component template

Commit:
Feature(player-indicator): Add an icon to the player template to show the users who they are

Commit:
Feature(player-indicator): Add an icon to the player template to show the users who they are

Commit:
Feature(add-question-timer-compnent): Add the timer and question component to the directive file

Commit:
Feature(add-question-timer-compnent): Add the timer and question component to the directive file

Commit:
Feature(refactor): Add spaces between divs to help in unerstanding what's going on

Commit:
Feature(refactor): Add spaces between divs to help in unerstanding what's going on

Commit:
Feature(rename): Rename timer.html to timer.tpl.html

Commit:
Feature(rename): Rename timer.html to timer.tpl.html

Commit:
Feature(leave-game): Add ng-click directive to leave game button

Commit:
Feature(leave-game): Add ng-click directive to leave game button

Commit:
Feature(increase-black-card-size): Increase the size of the black card to make question easy to read

Commit:
Feature(increase-black-card-size): Increase the size of the black card to make question easy to read

Commit:
Feature(chat-section): Modify game.tpl.html to make allowances for the chat functionality

Commit:
Feature(chat-section): Modify game.tpl.html to make allowances for the chat functionality

Commit:
Feature(show-question): Show questions on the black card properly

Add ngSanitize to the project so it can safely display the html
tags in the question

Commit:
Feature(show-question): Show questions on the black card properly

Add ngSanitize to the project so it can safely display the html
tags in the question

Commit:
Feature(change-timeout): Change the value of the waiting time for first player to start game

Commit:
Feature(change-timeout): Change the value of the waiting time for first player to start game

Commit:
Feature(answer-tag): Add the answer custom tag to the game screen

Commit:
Feature(answer-tag): Add the answer custom tag to the game screen

Commit:
Feature(answer-component): Implement the directive for the answer component

Commit:
Feature(answer-component): Implement the directive for the answer component

Commit:
Feature(text-style): Add styling to the text headers in the answer component

Commit:
Feature(text-style): Add styling to the text headers in the answer component

Commit:
Feature(chat-component): Add a directive for the chat feature

Commit:
Feature(chat-component): Add a directive for the chat feature

Commit:
Feature(chat-component): Add a directive for the chat feature

Commit:
Feature(chat-component): Add a directive for the chat feature

Commit:
Merge branch 'feature/#130341359/re-design-ui-for-gaming-screen' of github.com:andela/project-kinsan into feature/#130341359/re-design-ui-for-gaming-screen

Commit:
Feature(add-info-modal): Add a modal to show game information

Commit:
Feature(add-info-modal): Add a modal to show game information

Commit:
Feature(modal-button): Add buttons to go to homepage and play another game to the info modal

Commit:
Feature(modal-button): Add buttons to go to homepage and play another game to the info modal

demiladebamgbose committed:
Test(Timer test): Test timer component

Add tests for timer component

Finishes #130341359

demiladebamgbose committed:
Test(Timer test): Test timer component

Add tests for timer component

Finishes #130341359

demiladebamgbose committed:
Test(Question test): Test question component

Add tests for question component

Finishes #130341359

demiladebamgbose committed:
Test(Question test): Test question component

Add tests for question component

Finishes #130341359

Commit:
Feature(remove-unused-var): Removed the unused variables as noted by houndci

Commit:
Feature(remove-unused-var): Removed the unused variables as noted by houndci

Commit:
Feature(hide-players): Hide the players when the game has ended

Commit:
Feature(hide-players): Hide the players when the game has ended

demiladebamgbose committed:
Bug(js2html): Fix karma configuration

Add plugins to karma config file

Finishes #130341359

demiladebamgbose committed:
Bug(js2html): Fix karma configuration

Add plugins to karma config file

Finishes #130341359

demiladebamgbose committed:
Bug(Fix build on travis): Load plugins for karma

Add plugins to karma config file

Finishes #130341359

demiladebamgbose committed:
Bug(Fix build on travis): Load plugins for karma

Add plugins to karma config file

Finishes #130341359

Commit:
Feature(add-game-sass-file): Convert game css to sass

Commit:
Feature(add-game-sass-file): Convert game css to sass

demiladebamgbose committed:
Bug(Fix build on tavis CI): Remove ng-html2js from plugins

Remove ng-html2js from karma config file

Finishes #130341359

demiladebamgbose committed:
Bug(Fix build on tavis CI): Remove ng-html2js from plugins

Remove ng-html2js from karma config file

Finishes #130341359

Commit:
Merge branch 'feature/#130341359/re-design-ui-for-gaming-screen' of github.com:andela/project-kinsan into feature/#130341359/re-design-ui-for-gaming-screen

Commit:
Merge branch 'feature/#130341359/re-design-ui-for-gaming-screen' of github.com:andela/project-kinsan into feature/#130341359/re-design-ui-for-gaming-screen

Commit:
Feature(add-karma-ng-html2js-preprocessor): Add karma-ng-html2js-preprocessor to the project

Commit:
Feature(add-karma-ng-html2js-preprocessor): Add karma-ng-html2js-preprocessor to the project

Commit:
Feature(pop-up-implementation): Change the implementation of the popup that shows information

Commit:
Feature(pop-up-implementation): Change the implementation of the popup that shows information

Commit:
Feature(add-game-link): Add link to game screen in the landing page

Commit:
Feature(add-game-link): Add link to game screen in the landing page

Commit:
Feature(game-screen-reload): Add a reload for the game screen to refresh state

Commit:
Feature(game-screen-reload): Add a reload for the game screen to refresh state

demiladebamgbose committed:
Bug:(Fix hound issues): Fix issues raised by hound CI

Fix issues raised by hound CI by disabling eslint

Finishes #130341359

demiladebamgbose committed:
Bug:(Fix hound issues): Fix issues raised by hound CI

Fix issues raised by hound CI by disabling eslint

Finishes #130341359

demiladebamgbose committed:
Bug(fix hound issues): Fix hound issues

Fix hound complains in sass file

Finishes #130341359

demiladebamgbose committed:
Bug(fix hound issues): Fix hound issues

Fix hound complains in sass file

Finishes #130341359

demiladebamgbose committed:
Bug(Fix hound Issues): Fix hound CI issues

Fix issues raised by hound in sass file

Fixes #130341359

demiladebamgbose committed:
Bug(Fix hound Issues): Fix hound CI issues

Fix issues raised by hound in sass file

Fixes #130341359

ruqoyyasadiq commented on public/css/game.scss:
```diff
@@ -0,0 +1,185 @@
+  height: 50px;
+  line-height: 50px;
+  background-color: #4074f5;
+  color: white;
```
@demiladebamgbose @andela-oolutola use variables or just write the hex code not color names. You may want to do something in the line of `$white: #ffffff` and use the variable `$white` in all places where you'd like to make reference to the color.

ruqoyyasadiq replied:
same as above

ruqoyyasadiq replied:
same as above

demiladebamgbose committed:
Refctor(Fix hound issues): Refactor to fix hound issues

Refactor sass file to fix hound issues

Finishes #130341359

demiladebamgbose committed:
Refctor(Fix hound issues): Refactor to fix hound issues

Refactor sass file to fix hound issues

Finishes #130341359

ruqoyyasadiq commented on public/css/game.scss:
```diff
@@ -0,0 +1,186 @@
+  border-left: 3px solid #eee;
+}
+
+.players-bar {}
```
why do you have a class selector with an empty style? do remove it if you are not adding styles for that class. ;)

ruqoyyasadiq commented on public/css/game.scss:
```diff
@@ -0,0 +1,186 @@
+  }
+}
+
+.start-game, .awaiting-players {
```
What hound is saying is that you should do something like this...

```
.start-game,
.awaiting-players {
      style-name: value;
      style-name2: value2;
}
```

ruqoyyasadiq commented on public/css/game.scss:
```diff
@@ -0,0 +1,186 @@
+    width: 25%;
+    text-decoration: none;
+    height: 1em;
+    border-bottom: 0.125em solid white;
```
use variables for color hex codes.

ruqoyyasadiq commented on public/css/game.css:
```diff
@@ -0,0 +1,141 @@
+.container {
+  width: 100%;
+  margin: 0 !important; }
```
why are you using `!important`? You may want to try the concept of `specificity in css` over `!important rule`. In this case, having an `id` attached to the element in question and then adding styles to id selector makes the rule apply rather than using `!important` in that class.

demiladebamgbose committed:
Refactor(Fix hound issues): Refactor to shut hound up

Fix hound issues in sass file

Finishes #130341359

demiladebamgbose committed:
Refactor(Fix hound issues): Refactor to shut hound up

Fix hound issues in sass file

Finishes #130341359

ruqoyyasadiq commented on public/css/game.css:
```diff
@@ -0,0 +1,141 @@
+  height: 500px; }
+
+.deck-of-black-cards {
+  box-shadow: -2px -2px 0 0  #000000;
```
instead of 0px, write 0...so, you should have something like `-2px -2px 0 0 $color-variable`

ruqoyyasadiq commented on public/css/game.scss:
```diff
@@ -0,0 +1,186 @@
+  height: 200px;
+  width: 150px;
+  font-size: medium;
+  u {
```
have a line break after line 38

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

demiladebamgbose committed:
Refactor(scss-hound-correction): Make corrections on scss suggested by houndci

cyrielo committed:
#130341363 Add In-game chat feature (#24)

* chore(readme-update): update read me with new relevant info
Finishes #130373385

* feature(chat): in game chat

- Add branch for creating in game chat.

Start #130341363

* feature(chat):in game chat

Includes firebase reference and setup in foot.jade
add reference to css file in head.jade
add router to redirect to the game chat screen
add a game-chat controller
add a game chat service
add test for the game-chat

Start #130341363

* feature(chat): in game chat
- Add images for the chat widget
- Correct padding for elenents of the chat

#130341363

* feature(chat): in game chat
- Integrate front end with controller
- Implment <user> is typing feature
- Implement auto scroll to the newest chat
#130341363

* feature(chat): in game chat
-Remove border from message input in chat window
#130341363

* feature(chat): in game chat

-Update foot.jade to reference the firebase lib
-update bower dependency to reference firebase
-update karma.conf.js to be aware of firebase lib

Start #130341363

* feature(chat): in game

Refactor game-chat controller and integrate it to frontend
Refactor gamechat factory, add isTyping functionality
Complete writing tests for game-chat controller and game-chat factory

Start #130341363

* feature(chat): in game

-Resolve conflict in css file
-Update game.tpl.html to remove Doctype template
-Update game-chat.js controller to send message when enter key is pressed

Start #130341363

* feature(chat): in game chat

-Fix broken package.json
Start #130341363

* feature(chat): in game chat

Add fix for travis build failure

130341363

* feature(chat): in game chat

Fix travis build failure

130341363

* feature(chat): in game chat

Resolve HoundCI reported issues

Start 130341363

* feature(chat): in-game-chat

- Convert css to sass/scss
- add folder to gulp task

#130341363

* merge

* feature(chat): in-game-chat

- Make chat fixed with no retracting feature.
- Change color to blue

#130341363

* feature(chat): in-game-chat

- Make changes to pass hound test.

#130341363

* Hound Changes

* Fix Chat Text

* feature(chat): in game chat

Update karma.conf.js to load socket.io.js
Update app.js to add mean.system as dependency
Add socket.io.js library

130341363

* feature(chat): in game chat

Separate controller test and service test

130341363

* feature(chat): in game chat

Remove hardcoded values in game-chat controller
Add isTyping feature to controller
Add Jquery code to controller
Fix typo error in game-chat service
Remove jquery code from game.tpl.html

130341363

* feature(chat): in game chat

Mock the game service in the game-chat-ctrl.test.js
Remove the socket.io.js library

130341363

* feature(game): in-game-chat
- Remove css files from repo and from being tracked.
- Create seperate scss file for variables called variables.scss

#130341363

* feature(game): in-game-chat
- Make Hound Changes.

#130341363

ruqoyyasadiq commented:
@demiladebamgbose and @andela-oolutola fix conflicts

Commit:
Feature(correct-sass): Corrected an error which didn't make chat.scss see variable.scss

Commit:
Merge branch 'feature/#130341359/re-design-ui-for-gaming-screen' of github.com:andela/project-kinsan into feature/#130341359/re-design-ui-for-gaming-screen

Commit:
Refactor(hound): Make changes as noted by houndci

adetola-ralph commented:
@andela-rsadiq please don't merge in the pull request yet, we are still yet to fully integrate the chat features into the game screen

Commit:
Feature(game-id): Added the capability to generate different ids for games

Commit:
Feature(add-module): Inject mean.gameChat module into mean.directives

Commit:
Feature(chat-component): Converted the chat feature into a service

Commit:
Feature(chat-scss): Modify chat scss to make it fit into the page

Commit:
Feature(chat-template): Add the chat html to the chat template file

Commit:
Feature(html-structure): Modify game template file to have css structure divs around the template tags

Commit:
Feature(game-controller): Modify game controller to show chat screen on game start

Commit:
Feature(game-chat-service): Fix error in game-chat service which made the chat not store properly on firebase

Commit:
Refactor(game-chat-controller): Removed the game-chat controller as it was not necessary

Commit:
Feature(tests): Modify tests to pass

Commit:
Refactor(houndci): make changes to scss files suggested by hound

ruqoyyasadiq commented:
@demiladebamgbose @andela-oolutola can this PR be closed seeing that the same work was duplicated on another branch which has been merged into develop?
